tests/ide/ide_test.py

```diff
@@ -80,10 +80,20 @@ def test_multiple_errors() -> None:
             f'{path}:103:26: model: error: size of opaque field "Data" not multiple of 8 bit'
             " (Length -> Data)",
             f'{path}:106:4: parser: error: undefined type "RFLX_Invalid::PDU1" in refinement',
+            f'{path}:106:40: parser: error: undefined type "RFLX_Invalid::SDU2" in refinement '
+            'of "RFLX_Invalid::PDU1"',
             f'{path}:109:4: parser: error: undefined type "RFLX_Invalid::PDU1" in refinement',
+            f'{path}:109:26: parser: error: undefined type "RFLX_Invalid::SDU2" in refinement '
+            'of "RFLX_Invalid::PDU1"',
             f'{path}:112:4: parser: error: undefined type "RFLX_Invalid::Undef_PDU" in refinement',
+            f'{path}:112:31: parser: error: undefined type "RFLX_Invalid::R" in refinement '
+            'of "RFLX_Invalid::Undef_PDU"',
             f'{path}:115:4: parser: error: undefined type "RFLX_Invalid::PDU1" in refinement',
+            f'{path}:115:26: parser: error: undefined type "RFLX_Invalid::Undef_Type" in '
+            'refinement of "RFLX_Invalid::PDU1"',
             f'{path}:118:4: parser: error: undefined type "RFLX_Invalid::PDU1" in refinement',
+            f'{path}:118:33: parser: error: undefined type "RFLX_Invalid::R" in refinement '
+            'of "RFLX_Invalid::PDU1"',
```

Comment:
The parser is now stricter in which types he drops directly due to errors, and thus we get more "undefined type" errors. Wouldn't it be better to be more tolerant during the collection of types to prevent such subsequent parser errors?

Reply:
These additional messages are not because we detect more invalid types, but because we don't just call `fail` when the first undefined type is found in refinements. I agree that this is not ideal - we could consider returning some "invalid type" object instead of omitting the type when we find an error. This way it would not be counted as missing. But that's certainly nothing for this PR...

Reply:
I see. Then let's keep it as it is for now.